ckanext/dcatapchharvest/profiles.py

```diff
@@ -1036,21 +1037,20 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
     def _rights_and_license_to_graph(self, resource_dict, distribution):
         g = self.g
         if resource_dict.get('rights'):
-            rights_uri = dh.get_license_uri_by_name(
+            rights_uri = dh.get_license_ref_uri_by_homepage_uri(
                 resource_dict.get('rights')
             )
             if rights_uri is not None:
                 rights_ref = URIRef(rights_uri)
                 g.add((rights_ref, RDF.type, DCT.RightsStatement))
                 g.add((distribution, DCT.rights, rights_ref))
             if rights_uri is None:
-                rights_name = dh.get_license_name_by_uri(
+                rights_name = dh.get_license_name_by_homepage_uri(
                     resource_dict.get('rights')
                     )
                 if rights_name is not None:
-                    resource_rights_ref = URIRef(
-                        resource_dict.get('rights')
-                        )
+                    resource_rights_ref = dh.get_license_ref_uri_by_name(
+                        rights_name)
                     g.add((
                         resource_rights_ref,
                         RDF.type,
```

Comment:
I propose combining the logic for getting the `rights_uri` (here and in lines 1062-1078 below) into one helper function, which would do something like this:

- try getting the license uri from the mapping using the homepage uri
- if successful, return it
- get the license name from the mapping using the homepage uri
- get the license uri from the mapping using the name
- if successful, return it
- otherwise, return `None`

If we have a license/rights uri, we can add it to the graph here.

My reasoning is that we are not using those new helper functions (`get_license_ref_uri_by_homepage_uri`, etc.) anywhere else but here, so we might as well abstract the logic there to have it all in one place and make the method here more readable.

It would also be more consistent to always get a uri (or a URIRef, I guess) from those helpers - here it seems like we are getting a uri from `get_license_ref_uri_by_homepage_uri` and a URIRef from `get_license_ref_uri_by_name`, which is surprising.